text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,415 @@
+
+Note the separation of the namespace and name using a ':' is intentional. It is neither a valid [tag name](https://html.spec.whatwg.org/multipage/syntax.html#syntax-tag-name) nor a valid [custom element name](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) which should clearly signal that it is a special element (which will not be rendered to the DOM).
+
+Additionally, `<lwc:component>` is intended to be used in tandem with `lwc:is`, otherwise `<lwc:component>` serves no purpose. The compiler will report an error when `<lwc:component>` is used without the `lwc:is` template directive.
+
+##### `lwc:is`
+
+A new template directive called `lwc:is` will be used as a signal to the compiler that the constructor is unknown at compile time. 
+
+The directive has the following properties:
+
+1. It can only be used with `<lwc:component>` tag.
+    - An error will be issued when `lwc:is` is used on any other element.
+2. At compile time the value supplied to `lwc:is` can only be an expression.
+3. At runtime the expression supplied to `lwc:is` must be a `LightningElementConstructor`.
+    - An error will be thrown at runtime if anything other than a `LightningElementConstructor` is passed to `lwc:is`.
+
+#### Instantiation of the custom element
+
+Dynamic component instantiation follows the same creation path as any other LWC custom element. When the dynamic component is ready to be rendered, the element will be created and mounted to the DOM.  If the component constructor supplied to `lwc:is` changes, the existing element will be unmounted.
+
+This means that when the value provided to `lwc:is` changes the component and all of its children will be destroyed and recreated using the new constructor.  In addition, when the value provided to `lwc:is` is evaluated, the LWC engine will render depending on the following scenarios:
+
+- **The constructor is falsy**
+    - When the constructor is falsy the `<lwc:component>` tag along with all of its children are not rendered to the DOM.
+- **The constructor is defined and is not a `LightningElementConstructor`**
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc:component>`. The tag name used for the dynamic component is the value returned from `getComponentRegisteredName`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
+
+#### Selecting the dynamic component
+
+Dynamic components have certain properties that will not work with `this.template.querySelector`:
```

Comment:
this paragraph and list below are confusing. The idea is probably to explain that you don't have a name to query for... so either you add a class to the `<lwc:component>`, so you can query with that, or you use a ref, which is the recommended way. It is not really about `<lwc:component>` since they will never see that in the dom, so no need to mention it.

Reply:
That's what I was going for, I'll reword this portion and make it more clear.